sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RetriableResponseWorkItem.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation;
+
+import org.apache.qpid.proton.amqp.transport.DeliveryState;
+import reactor.core.publisher.MonoSink;
+
+import java.time.Duration;
+
+/**
+ * Work item which returns {@link DeliveryState} to subscriber and that can be scheduled multiple times..
+ */
+final class RetriableResponseWorkItem extends WorkItem {
```

Comment:
I'm not convinced we need this class, why not change them all to use a `Mono<DeliveryState>` and in the latter case, it completes with empty.